packages/adaptive_scaffold/example/lib/adaptive_layout_demo.dart

```diff
@@ -54,54 +54,200 @@ class MyHomePage extends StatelessWidget {
+              onPressed: () {},
+              icon: const Icon(Icons.folder_copy_outlined),
+              iconSize: 21,
+            ),
+            const SizedBox(
+              width: 21,
+            ),
+            const Flexible(
+                child: Text(
+              "Receipts",
+              overflow: TextOverflow.ellipsis,
+            )),
+          ],
+        ),
+      ],
+    );
+
     return AdaptiveLayout(
-      // Primary navigation config has nothing from 0 to 600 dp screen width,
-      // then an unextended NavigationRail with no labels and just icons then an
-      // extended NavigationRail with both icons and labels.
       primaryNavigation: SlotLayout(
-        config: <Breakpoint, SlotLayoutConfig?>{
+        config: {
+          Breakpoints.small: SlotLayout.from(
+              key: const Key('pnav'), builder: (_) => const SizedBox.shrink()),
           Breakpoints.medium: SlotLayout.from(
             inAnimation: AdaptiveScaffold.leftOutIn,
-            key: const Key('pn'),
-            builder: (_) =>
-                AdaptiveScaffold.toNavigationRail(destinations: destinations),
+            key: const Key('pnav1'),
+            builder: (_) => AdaptiveScaffold.toRailFromDestinations(
+                leading: const Icon(Icons.menu), destinations: destinations),
           ),
           Breakpoints.large: SlotLayout.from(
             key: const Key('pn1'),
             inAnimation: AdaptiveScaffold.leftOutIn,
-            builder: (_) => AdaptiveScaffold.toNavigationRail(
-                extended: true, destinations: destinations),
+            builder: (_) => AdaptiveScaffold.toRailFromDestinations(
+              extended: true,
+              leading: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceAround,
+                children: const [
+                  Text(
+                    "REPLY",
+                    style: TextStyle(color: Color.fromARGB(255, 255, 201, 197)),
+                  ),
+                  Icon(Icons.menu_open)
+                ],
+              ),
+              destinations: destinations,
+              trailing: trailingNavRail,
+            ),
           ),
         },
       ),
-      // Body switches between a ListView and a GridView from small to medium
-      // breakpoints and onwards.
       body: SlotLayout(
-        config: <Breakpoint, SlotLayoutConfig?>{
+        config: <Breakpoint, SlotLayoutConfig>{
           Breakpoints.small: SlotLayout.from(
             key: const Key('body'),
             builder: (_) => ListView.builder(
-                itemCount: 10, itemBuilder: (_, int idx) => children[idx]),
+              itemCount: 10,
+              itemBuilder: (context, index) => Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: Container(
+                  color: const Color.fromARGB(255, 255, 201, 197),
+                  height: 400,
+                ),
+              ),
+            ),
```

Comment:
You already have the children variable defined so it would make more sense to use it here as it is used in the previous version rather than recreating the widget.